clang/lib/3C/RewriteUtils.cpp

```diff
@@ -593,7 +593,7 @@ void RewriteConsumer::emitRootCauseDiagnostics(ASTContext &Context) {
         // or are in the main file of the TU. Alternatively, don't filter causes
         // if -warn-all-root-cause is passed.
         int PtrCount = I.getNumPtrsAffected(WReason.first);
-        if (_3COpts.WarnAllRootCause || SM.isInMainFile(SL) || PtrCount > 1) {
+        if (_3COpts.WarnAllRootCause || SM.isInMainFile(SL) || PtrCount > 0) {
```

Comment:
Please update the comment to match.

Reply:
Also, would it make sense to remove the `SM.isInMainFile(SL)` disjunct now?  (Can there be a root cause in the main file with zero affected pointers?)